src/sinks/statsd/encoder.rs

```diff
@@ -0,0 +1,355 @@
+use std::{fmt::Display, io::Write};
+
+use bytes::{BufMut, BytesMut};
+use tokio_util::codec::Encoder;
+use vector_core::event::{Metric, MetricKind, MetricTags, MetricValue, StatisticKind};
+
+use crate::{
+    internal_events::StatsdInvalidMetricError,
+    sinks::util::{buffer::metrics::compress_distribution, encode_namespace},
+};
+
+#[derive(Debug, Clone)]
+pub struct StatsdEncoder {
+    default_namespace: Option<String>,
+}
+
+impl StatsdEncoder {
```

Comment:
I expect one day we'll still want to move this into `lib/codecs` proper, yeah?

Reply:
Probably?

I'm not sure we'd want to throw it into the main encoding/decoding enums -- like, is there really value in letting arbitrary sinks emit statsd-encoded metrics? etc -- but it could certainly live in `codecs` for the purpose of being isolated, better tested, and colocated with other similar code.